lib/src/command/run.dart

```diff
@@ -86,4 +91,98 @@ class RunCommand extends PubCommand {
     });
     await flushThenExit(exitCode);
   }
+
+  /// Implement a v2 mode for use in `dartdev run`.
+  ///
+  /// Usage: `dartdev run [package[:command]]`
+  ///
+  /// If `package` is not given, defaults to current root package.
+  /// If `command` is not given, defaults to name of `package`.
+  /// If neither `package` or `command` is given and `command` with name of
+  /// the current package doesn't exist we fallback to `'main'`.
+  ///
+  /// Runs `bin/<command>.dart` from package `<package>`, with snapshot stored
+  /// in `.dart_tool/pub/bin/<package>/<command>.dart.snapshot.dart2`.
+  Future _runV2() async {
+    var package = entrypoint.root.name;
+    var command = package;
+    var args = <String>[];
+
+    if (argResults.rest.isNotEmpty) {
+      if (argResults.rest[0].contains('/')) {
```

Comment:
there are likely many characters it can't contain - why is `/` special? Do we need to check for `\` on windows?

Reply:
We could probably check more. We could disallow all characters not allowed in filenames, but that's probably not necessary as checking if a file with an illegal filename exists hopefully returns false.

But checking for slash and backslash is nice, that way we avoid going into sub-folders by accident.